src/Entity/ListBuilder/SubscriptionListBuilder.php

```diff
@@ -248,10 +248,56 @@ public function buildRow(EntityInterface $entity) {
    * {@inheritdoc}
    */
   public function render() {
-    $build = parent::render();
+    $build = [];
 
-    array_push($build['table']['#cache']['contexts'], 'url.query_args');
-    array_push($build['table']['#cache']['tags'], 'apigee_my_subscriptions');
+    // Group the subscriptions by status.
+    $subscriptions = [
+      SubscriptionInterface::STATUS_ACTIVE => [
+        'title' => $this->t('Active and Future Plans'),
+      ],
+      SubscriptionInterface::STATUS_ENDED => [
+        'title' => $this->t('Cancelled and Expired Plans'),
+      ],
+    ];
+
+    /** @var \Drupal\apigee_m10n\Entity\SubscriptionInterface $entity */
+    foreach ($this->load() as $entity) {
+      if (in_array($entity->getSubscriptionStatus(), [SubscriptionInterface::STATUS_ACTIVE, SubscriptionInterface::STATUS_FUTURE])) {
+        $subscriptions[SubscriptionInterface::STATUS_ACTIVE]['entities'][] = $entity;
+        continue;
+      }
+
+      $subscriptions[SubscriptionInterface::STATUS_ENDED]['entities'][] = $entity;
+    }
+
+    foreach ($subscriptions as $key => $subscription) {
+      $build[$key]['heading'] = [
+        '#type' => 'html_tag',
+        '#value' => $subscription['title'],
+        '#tag' => 'h3',
+      ];
+
+      $build[$key]['table'] = [
+        '#type' => 'table',
+        '#header' => $this->buildHeader(),
+        '#title' => $subscription['title'],
+        '#rows' => [],
+        '#empty' => $this->t('There are no @label yet.', ['@label' => $this->entityType->getPluralLabel()]),
+        '#cache' => [
+          'contexts' => $this->entityType->getListCacheContexts(),
+          'tags' => $this->entityType->getListCacheTags(),
+        ],
+      ];
+
+      foreach ($subscription['entities'] as $entity) {
```

Comment:
I'm getting a couple of messages on this (notice + warning) on this:

`Notice: Undefined index: entities in Drupal\apigee_m10n\Entity\ListBuilder\SubscriptionListBuilder->render() (line 292 of modules/contrib/apigee_m10n/src/Entity/ListBuilder/SubscriptionListBuilder.php).`

```
Drupal\apigee_m10n\Entity\ListBuilder\SubscriptionListBuilder->render() (Line: 60)
Drupal\apigee_m10n\Entity\ListBuilder\SubscriptionListBuilderForDeveloper->render(Object)
call_user_func_array(Array, Array) (Line: 123)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 582)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 124)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->wrapControllerExecutionInRenderContext(Array, Array) (Line: 97)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 151)
Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object, 1) (Line: 68)
Symfony\Component\HttpKernel\HttpKernel->handle(Object, 1, 1) (Line: 57)
Drupal\Core\StackMiddleware\Session->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object, 1, 1) (Line: 106)
Drupal\page_cache\StackMiddleware\PageCache->pass(Object, 1, 1) (Line: 85)
Drupal\page_cache\StackMiddleware\PageCache->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object, 1, 1) (Line: 52)
Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object, 1, 1) (Line: 23)
Stack\StackedHttpKernel->handle(Object, 1, 1) (Line: 693)
Drupal\Core\DrupalKernel->handle(Object) (Line: 19)
```

--- 

`Warning: Invalid argument supplied for foreach() in Drupal\apigee_m10n\Entity\ListBuilder\SubscriptionListBuilder->render() (line 292 of modules/contrib/apigee_m10n/src/Entity/ListBuilder/SubscriptionListBuilder.php).`

```
Drupal\apigee_m10n\Entity\ListBuilder\SubscriptionListBuilder->render() (Line: 60)
Drupal\apigee_m10n\Entity\ListBuilder\SubscriptionListBuilderForDeveloper->render(Object)
call_user_func_array(Array, Array) (Line: 123)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 582)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 124)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->wrapControllerExecutionInRenderContext(Array, Array) (Line: 97)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 151)
Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object, 1) (Line: 68)
Symfony\Component\HttpKernel\HttpKernel->handle(Object, 1, 1) (Line: 57)
Drupal\Core\StackMiddleware\Session->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object, 1, 1) (Line: 106)
Drupal\page_cache\StackMiddleware\PageCache->pass(Object, 1, 1) (Line: 85)
Drupal\page_cache\StackMiddleware\PageCache->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object, 1, 1) (Line: 52)
Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object, 1, 1) (Line: 23)
Stack\StackedHttpKernel->handle(Object, 1, 1) (Line: 693)
Drupal\Core\DrupalKernel->handle(Object) (Line: 19)
```

Reply:
I'll fix this. Good catch. Thanks @jacine